src/Mailer/Message/CitizenActionCancellationMessage.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace AppBundle\Mailer\Message;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\CitizenAction;
+use AppBundle\Entity\EventRegistration;
+use Ramsey\Uuid\Uuid;
+
+final class CitizenActionCancellationMessage extends Message
+{
+    public static function create(
+        array $recipients,
+        Adherent $author,
+        CitizenAction $citizenAction,
+        \Closure $recipientVarsGenerator
```

Comment:
This is fragile and un-needed (Or perhaps I am missing something :D)
Just enforce the 1st argument to be an `EventRegistration`. It's the job of the caller to do the `array_shift` stuff.

Reply:
About the closure generating the recipient vars: actually this is not needed and will be refactored in the `email-templates` PR, i suggest you leave this as is in this PR.